formats/dsd/http.go

```diff
@@ -0,0 +1,132 @@
+package dsd
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"strings"
+)
+
+var (
+	ErrMissingBody        = errors.New("dsd: missing http body")
+	ErrMissingContentType = errors.New("dsd: missing http content type")
+)
+
+const (
+	httpHeaderContentType = "Content-Type"
+)
+
+func LoadFromHTTPRequest(r *http.Request, t interface{}) (format uint8, err error) {
+	if r.Body == nil {
+		return 0, ErrMissingBody
+	}
+	defer r.Body.Close()
```

Comment:
since this is used on the server side there is no need to check if `r.Body` is nil. Also, IMHO `LoadFromHTTPRequest` should NOT close the response body because that's out of it's responsibility. 

To cite the docs:
```
	// For server requests, the Request Body is always non-nil
	// but will return EOF immediately when no body is present.
	// The Server will close the request body. The ServeHTTP
	// Handler does not need to.

```